### PR TITLE
Support multiple resources per table

### PR DESCRIPTION
**Pull request by camertron**
Currently txdb assumes there will always be a 1:1 relationship between tables and Transifex resources. This turned out to be an incorrect assumption. We'd like to submit one resource per email template, but all our email strings are stored in a single database table. This PR allows a database table to map to more than one resource. To summarize the changes:
1. The two backend interface methods - `#read_content_from` and `#write_content_to` - have been modified to accept and return `TxResource` objects instead of serialized YAML strings. `#read_content_from` actually returns an array of resources, enabling more than one resource per table.
2. `Txdb::TxResource` wraps an instance of `Txgh::TxResource` up with a string of serialized YAML (or whatever format) so they can be passed around together.

I also think this requirement has resulted in better abstractions in general, so yay for that.

@lumoslabs/platform

### CODE REVIEW

File: lib/txdb/backends/globalize/reader.rb [hunk 1]
```diff
@@ -12,11 +14,42 @@ def initialize(table)
...
 
         def read_content
-          { origin_table_name(table.name) => content_for_records }
+          [Txdb::TxResource.new(resource, serialized_content)]
         end
 
         private
 
+        def resource
+          @resource ||= Txgh::TxResource.new(
```
Comment (zvkemp) [inline lib/txdb/backends/globalize/reader.rb:23]: Correct namespace? Txgh instead of Txdb?

File: lib/txdb/handlers/hook_handler.rb [hunk 2]
```diff
@@ -19,35 +19,66 @@ def initialize(request)
...
       def handle
-        tables.each do |table|
-          content = table.database.transifex_api.download(table.resource, locale)
-          table.write_content(content, locale)
+        if authentic_request?
+          downloader.download_resource(resource, table, locale)
+          respond_with(200, {})
+        else
+          respond_with_error(401, 'Unauthorized')
         end
```
Comment (zvkemp) [inline lib/txdb/handlers/hook_handler.rb]: prefer guard clause over `if... else`

File: lib/txdb/handlers/hook_handler.rb [hunk 3]
```diff
@@ -19,35 +19,66 @@ def initialize(request)
...
 
       def handle
-        tables.each do |table|
-          content = table.database.transifex_api.download(table.resource, locale)
-          table.write_content(content, locale)
+        if authentic_request?
+          downloader.download_resource(resource, table, locale)
+          respond_with(200, {})
+        else
+          respond_with_error(401, 'Unauthorized')
```
Comment (zvkemp) [inline lib/txdb/handlers/hook_handler.rb]: prefer guard clause over if... else

File: lib/txdb/handlers/triggers/handler.rb [hunk 4]
```diff
@@ -24,26 +24,17 @@ def handle_safely
...
-        def databases
-          @databases ||= if database_name = request.params['database']
-            Txdb::Config.databases.select do |database|
-              database.database == database_name
-            end
-          else
-            Txdb::Config.databases
+        def database
+          @database ||= Txdb::Config.databases.find do |database|
+            database.database == request.params['database']
```
Comment (zvkemp) [inline lib/txdb/handlers/triggers/handler.rb:29]: more than 50% of these words is `database`

Comment (zvkemp) [inline lib/txdb/handlers/triggers/handler.rb:29]: Possible to rename `database.database` to `database.name`? Block param also has same name as `__method__`, maybe use `config` instead?

**Comment by zvkemp** [comment]
Seems pretty good; a few comments re: naming.

Might be useful to see a diagram of how all of this works

File: lib/txdb/backends/globalize/reader.rb (hunk 1 above)
Comment (camertron) [inline lib/txdb/backends/globalize/reader.rb:23]: Yeah, this is correct. Txdb and Txgh both define the concept of a resource. `Txdb::TxResource` actually wraps `Txgh::TxResource` together with some content.

File: lib/txdb/handlers/triggers/handler.rb (hunk 4 above)
Comment (camertron) [inline lib/txdb/handlers/triggers/handler.rb:29]: 😳  yeah, `database.name` would be way better for cases like this. I think that should be a separate PR tho.

**Comment by camertron** [comment]
Ok, so I kinda sorta did some UML:
txdb

Basically it all starts with a `Database` object, which contains a list of `Table` objects. `Database` also maintains a connection to a database via Sequel's adapter abstraction. Writing and reading data to/from the database is done via classes that implement the `Backend` interface. We make use of the `Globalize::Backend` to read and write records managed by globalize. Read operations return arrays of instances of `TxResource` and write operations receive instances of `TxResource`.

**Comment by zvkemp** [comment]
🆒 !
